### PR TITLE
Reworks staves, canes, sticks and brooms.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
## Description of changes
- Staves/sticks/canes/brooms use the single icon system and are greyscaled to use material color.
- The basic staff type is now a plain length of wood (or such) with no magical crystal.
- Brooms can be used to sweep dirt and some decals from turfs.
- Brooms can be crafted from a wooden staff and 5 dried grass.
- Staves and simple canes can be crafted from wood.
- Canes are no longer a staff subtype.
- Items now use a proc to provide stance support.

image

## TODO
- [x] Readd wizard staves from wizard modpack PR.
- ~~Update wizard staff check to allow fancy cane to work as a staff.~~ Staves aren't checked for spells??
- [x] Test basic item spawning/appearance.
- [x] Test sword cane interactions.

## Why and what will this PR improve
Allows mundane staves to be used for stuff.
Brooms do a thing now.

## Authorship
Myself.

## Changelog
:cl:
tweak: Brooms can be used to sweep up dirt.
/:cl:

### CODE REVIEW

**Comment by MistakeNot4892** [comment]
Depends on #4396

**Comment by MistakeNot4892** [comment]
Rebased to remove dep.

File: code/game/objects/items/weapons/broom.dm [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+/obj/item/staff/broom/resolve_attackby(atom/A, mob/user, click_params)
+
+	if(user.a_intent != I_HURT)
+
+		// Sweep up dirt.
+		if(isturf(A))
+			var/turf/cleaning = A
+			var/dirty = cleaning.get_dirt()
+			if(dirty)
```
Comment (out-of-phaze) [inline code/game/objects/items/weapons/broom.dm:48]: this means you can sweep a tile immediately after someone steps on it, which, while funny, is a bit obtrusive. do you have any opinion on adding a threshold check so it has to be greater than some certain dirtiness amount? regardless not a blocker

**Review by comma — APPROVED** [review]

File: code/modules/crafting/stack_recipes/recipes_struts.dm [hunk 2]
```diff
@@ -7,11 +7,26 @@
 	one_per_turf                = TRUE
 	on_floor                    = TRUE
 	difficulty                  = MAT_VALUE_HARD_DIY
-	available_to_map_tech_level = MAP_TECH_LEVEL_SPACE
+	available_to_map_tech_level = MAP_TECH_LEVEL_MEDIEVAL
```
Comment (out-of-phaze) [inline code/modules/crafting/stack_recipes/recipes_struts.dm:10]: huh. does this mean we can make table frames out of wood or something?

**Review by out-of-phaze — APPROVED** [review]

File: code/game/objects/items/weapons/broom.dm (hunk 1 above)
Comment (MistakeNot4892) [reply code/game/objects/items/weapons/broom.dm:48]: I should add a default threshold check for cleanbots too as they get excited to clean turfs as soon as someone steps on it.

File: code/modules/crafting/stack_recipes/recipes_struts.dm (hunk 2 above)
Comment (MistakeNot4892) [reply code/modules/crafting/stack_recipes/recipes_struts.dm:10]: Should let us make them out of bones.

**Pull request merged**